app/crud/wh_sched.py

```diff
@@ -286,155 +285,143 @@ def _make_cron_schedule(
     return f"{offset} {cron_hours} * * {days_of_week} America/Los_Angeles", True
 
 
-class WarehouseSchedulesTask:
-    """
-    WarehouseSchedulesTask encapsulates the logic to apply the warehouse schedules against the warehouses
-    in the Snowflake Account. The logic in this class is stateless and does not need the encapsulation that the class
-    provides; however, this class eases the injection of test data via unittest.mock.
+def get_last_run(session: Session) -> datetime.datetime:
+    return (
+        session.table("internal.task_warehouse_schedule")
+        .filter(col("success"))
+        .select(sp_max("run").as_("x"))
+        .collect()[0]
+        .X
+    )
 
-    TODO move the logic in this class to functions in this file and figure out how to updated the mocking invocations to
-         override the methods which execute queries against Snowflake.
-    """
 
-    task_name: ClassVar[str] = "TASKS.WAREHOUSE_SCHEDULING"
-    task_offsets: ClassVar[List[int]] = [0, 15, 30, 45]
+def get_schedules(session: Session, is_weekday: bool) -> pd.DataFrame:
+    return (
+        session.table("internal.wh_schedules")
+        .filter(col("weekday") == is_weekday)
+        .filter(col("enabled"))
+        .to_pandas()
+    )
 
-    @classmethod
-    def disable_all_tasks(cls, session: Session):
-        statements = [
-            f"alter task if exists {cls.task_name}_{offset} suspend;"
-            for offset in cls.task_offsets
-        ]
-        statements.insert(0, "begin")
-        statements.append("end;")
-        session.sql("\n".join(statements)).collect()
+
+task_name: ClassVar[str] = "TASKS.WAREHOUSE_SCHEDULING"
+task_offsets: ClassVar[List[int]] = [0, 15, 30, 45]
```

Comment:
Should these just be `str` and `List[int]` now that the `class` is removed?

Reply:
yup, fixed